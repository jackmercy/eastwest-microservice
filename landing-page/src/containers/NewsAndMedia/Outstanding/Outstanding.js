import { memo } from 'react';
import PropTypes from 'prop-types';

import { Content } from '../../components';

import Item from './Item';
import useStyles from './outstanding.styles';

const Outstanding = ({ data, categories }) => {
  const classes = useStyles();

  return data && data.length > 0 ? (
    <Content className={classes.root}>
      <div className={classes.main}>
        <Item
          data={{
            ...data[0],
            category: categories.find((c) => c.id === data[0]?.category),
          }}
          main
          className={classes.mainItem}
          dataTest="outstanding-main-item"
        />
      </div>
      <div className={classes.others}>
        {data.slice(1, 3).map((item) => (
          <Item
            key={item.id}
            data={{
              ...item,
              category: categories.find((c) => c.id === item.category),
            }}
            className={classes.otherItem}
            size="small"
            dataTest="outstanding-sub-item"
          />
        ))}
      </div>
    </Content>
  ) : null;
};
Outstanding.propTypes = {
  data: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      slug: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
      category: PropTypes.string.isRequired,
      thumbnail: PropTypes.string.isRequired,
    }),
  ),
  categories: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
    }),
  ),
};
Outstanding.defaultProps = {
  data: [],
  categories: [],
};

export default memo(Outstanding);
