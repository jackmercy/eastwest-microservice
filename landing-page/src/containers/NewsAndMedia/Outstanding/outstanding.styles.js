import { createUseStyles } from 'react-jss';

import { rem } from '../../../utils/jss';
import BREAKPOINTS from '../../../constants/BREAKPOINTS';

export default createUseStyles({
  root: {
    display: 'flex',
    boxSizing: 'border-box',
    marginBottom: rem(64),
    width: '100%',
    height: rem(566),
    transition: 'height .3s ease',
    overflow: 'hidden',
  },
  main: { flex: 2, marginRight: rem(32) },
  others: { flex: 1, display: 'flex', flexDirection: 'column' },
  mainItem: {},
  otherItem: {
    height: 'auto',
    maxHeight: '50%',
    flex: 1,
    '& + $otherItem': { marginTop: rem(32) },
  },

  [`@media only screen and (max-width: ${BREAKPOINTS.maxLg}px)`]: {
    root: { height: rem(450) },
    main: { marginRight: rem(24) },
    otherItem: { '& + $otherItem': { marginTop: rem(24) } },
  },
  [`@media only screen and (max-width: ${BREAKPOINTS.maxSm}px)`]: {
    root: { display: 'block', height: 'auto', marginBottom: rem(32) },
    main: { marginRight: 0, marginBottom: rem(32) },
    mainItem: { height: rem(231) },
    others: { display: 'block' },
    otherItem: {
      height: rem(231),
      maxHeight: '100%',
      '& + $otherItem': { marginTop: rem(32) },
    },
  },
});
