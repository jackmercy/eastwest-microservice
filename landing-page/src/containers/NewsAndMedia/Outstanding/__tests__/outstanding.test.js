import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Outstanding from '../Outstanding';

describe('Testing NewsAndMedia > Outstanding', () => {
  let wrapper;
  const props = {
    data: [
      {
        id: 'abc',
        slug: 'test-new',
        title: 'Some title',
        category: '610752dd1f174b00188e6359',
        thumbnail: 'image url',
        shortDescription: 'some description',
      },
      {
        id: 'abc1',
        slug: 'test-new',
        title: 'Some title',
        category: '610752dd1f174b00188e6359',
        thumbnail: 'image url',
        shortDescription: 'some description',
      },
      {
        id: 'abc2',
        slug: 'test-new',
        title: 'Some title',
        category: '610752dd1f174b00188e6359',
        thumbnail: 'image url',
        shortDescription: 'some description',
      },
    ],
    categories: [
      {
        tags: ['Insights'],
        name: 'Insights',
        slug: 'insights',
        shortDescription: 'Insights',
        id: '610752dd1f174b00188e6359',
      },
    ],
  };

  beforeAll(() => {
    wrapper = mount(<Outstanding {...props} />);
  });

  it('should render match snapshot', () => {
    const wrapper = shallow(<Outstanding {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should have a main item', () => {
    expect(wrapper.find('[data-test="outstanding-main-item"]')).toHaveLength(1);
  });

  it('should have 2 sub items', () => {
    expect(wrapper.find('[data-test="outstanding-sub-item"]')).toHaveLength(2);
  });
});
