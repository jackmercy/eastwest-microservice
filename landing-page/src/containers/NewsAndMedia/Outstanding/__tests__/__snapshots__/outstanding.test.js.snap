// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing NewsAndMedia > Outstanding should render match snapshot 1`] = `
<Memo(Content)
  className="root-0-2-1"
>
  <div
    className="main-0-2-2"
  >
    <Memo(Item)
      className="mainItem-0-2-4"
      data={
        Object {
          "category": Object {
            "id": "610752dd1f174b00188e6359",
            "name": "Insights",
            "shortDescription": "Insights",
            "slug": "insights",
            "tags": Array [
              "Insights",
            ],
          },
          "id": "abc",
          "shortDescription": "some description",
          "slug": "test-new",
          "thumbnail": "image url",
          "title": "Some title",
        }
      }
      dataTest="outstanding-main-item"
      main={true}
    />
  </div>
  <div
    className="others-0-2-3"
  >
    <Memo(Item)
      className="otherItem-0-2-5"
      data={
        Object {
          "category": Object {
            "id": "610752dd1f174b00188e6359",
            "name": "Insights",
            "shortDescription": "Insights",
            "slug": "insights",
            "tags": Array [
              "Insights",
            ],
          },
          "id": "abc1",
          "shortDescription": "some description",
          "slug": "test-new",
          "thumbnail": "image url",
          "title": "Some title",
        }
      }
      dataTest="outstanding-sub-item"
      key="abc1"
      size="small"
    />
    <Memo(Item)
      className="otherItem-0-2-5"
      data={
        Object {
          "category": Object {
            "id": "610752dd1f174b00188e6359",
            "name": "Insights",
            "shortDescription": "Insights",
            "slug": "insights",
            "tags": Array [
              "Insights",
            ],
          },
          "id": "abc2",
          "shortDescription": "some description",
          "slug": "test-new",
          "thumbnail": "image url",
          "title": "Some title",
        }
      }
      dataTest="outstanding-sub-item"
      key="abc2"
      size="small"
    />
  </div>
</Memo(Content)>
`;
